### PR TITLE
TaskScheduler API + demo pages (after rebase)

### PR DESCRIPTION
**Pull request by martap**

### CODE REVIEW

File: demo/calendar.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        font-weight:bold;
+}
+.days_from_other_month 
+{
+        color: #C0C0C0;
+}
+
+ td.days_from_other_month:hover
+{
+	background-color: white;
```
Comment (marcoscaceres) [inline demo/calendar.css]: Nit pick: Please align this line.

File: demo/calendar.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
+var calendar_date=new Date ();
```
Comment (marcoscaceres) [inline demo/calendar.js]: Your indenting is a messed up. Can I suggest you run your code through http://jsbeautifier.org/ . That will make it easier to review and maintain.

File: demo/calendar.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+calendar_date.setSeconds(0);
+
+function calendar(date)
+{
+        var day = date.getDate(),
+	    month = date.getMonth(),
+	    year = date.getUTCFullYear(),
+	    months = new Array('January', 'February', 'March', 'April', 'May', 'June', 'Jully', 'August', 'September', 'October', 'November', 'December'),
```
Comment (marcoscaceres) [inline demo/calendar.js]: Avoid using `new Array()` to construct arrays, instead, do this:

```
months = ['January', 'February', 'March', 'April', 'May', 'June', 'Jully', 'August', 'September', 'October', 'November', 'December'],
```

Comment (marcoscaceres) [inline demo/calendar.js]: This also applies to the following three lines and elsewhere.

File: demo/calendar.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	    days_in_month = new Array(31,28,31,30,31,30,31,31,30,31,30,31),
+	    days_in_week = new Array ('Sun','Mon','Tue','Wed','Thu','Fri','Sat'),
+	    month_days_number = null,
+	    first_day_of_month = date,
+	    week_number = 0,
+	    first_week = null,
+	    last_week_indicator = false;
+	
+	var today = day + ' ' + months[month] + ' ' + year;
```
Comment (marcoscaceres) [inline demo/calendar.js]: This line can be combined with the lines above (i.e., get rid of `var`).

File: demo/calendar.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+	/*
+	 * handle leap-year
+	 */
+	if(year%4 === 0 && year != 1900)
```
Comment (marcoscaceres) [inline demo/calendar.js]: Use "!==" instead.

File: demo/calendar.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
+var calendar_date=new Date ();
+calendar_date.setSeconds(0);
+
+function calendar(date)
+{
+        var day = date.getDate(),
+	    month = date.getMonth(),
+	    year = date.getUTCFullYear(),
+	    months = new Array('January', 'February', 'March', 'April', 'May', 'June', 'Jully', 'August', 'September', 'October', 'November', 'December'),
+	    days_in_month = new Array(31,28,31,30,31,30,31,31,30,31,30,31),
```
Comment (marcoscaceres) [inline demo/calendar.js]: I'm not sure if you need this... the logic is basically evens are 31 and odds are 30, unless it's the second month.

File: .gitignore [hunk 7]
```diff
@@ -4,3 +4,5 @@
 *.sublime-project
 
 *.sublime-workspace
+
+*.*~
```
Comment (marcoscaceres) [inline .gitignore:8]: This is ignoring any file that ends with a "~", right?

Comment (martap) [inline .gitignore:8]: Yes you are right. This is just to make my life easier - to ignore temporary files that are created on my system.

File: demo/calendar.js (hunk 6 above)
Comment (martap) [inline demo/calendar.js]: +august is exception. I'll check if change will look nicer :) thanks

File: demo/calendar.js (hunk 5 above)
Comment (martap) [inline demo/calendar.js]: oh.. I overlooked it. Thanks, I'll fix it.

File: demo/calendar.js (hunk 4 above)
Comment (martap) [inline demo/calendar.js]: ok

File: demo/calendar.js (hunk 3 above)
Comment (martap) [inline demo/calendar.js]: ok, I'll fix it, thanks :)

**Comment by martap** [comment]
@marcoscaceres I've fixed all issues you raised, I've also used jsbeautifier. Please check it.

**Comment by marcoscaceres** [comment]
Thanks for the update! I'm currently at Edgeconf in NYC, but will try to finish the review later this week.

**Comment by martap** [comment]
Sorry I've clicked "close" by mistake....

**Comment by marcoscaceres** [comment]
This looks good! We can probably clean up a few of the algorithms - but I think it's a good base for us to work from!

**Comment by martap** [comment]
Yupi!! :dancer:

**Pull request merged**